test/e2e/network/network_policy.go

```diff
@@ -532,12 +641,601 @@ var _ = SIGDescribe("NetworkPolicy", func() {
+			podServer, service = createPodWithClientServerAndService(f, f.Namespace, "pod-a", []int{80, 81}, "client-a", "pod-a-server-responded")
+
+			ginkgo.By("Waiting for pod ready", func() {
+				err := f.WaitForPodReady(podServer.Name)
+				framework.ExpectNoError(err, "Error occurred while waiting for pod type: Ready.")
+			})
+
+			// podServerLabelSelector holds the value for the podSever's label "pod-name".
+			podServerLabelSelector = podServer.ObjectMeta.Labels["pod-name"]
+
+			// Create pods, which should be able to communicate with the server on port 80 and 81.
+			ginkgo.By("Testing pods can connect to both ports when no policy is present.")
+			testCanConnect(f, f.Namespace, "client-can-connect-80", service, 80)
+			testCanConnect(f, f.Namespace, "client-can-connect-81", service, 81)
+		})
+
+		ginkgo.AfterEach(func() {
+			cleanupServerPodAndService(f, podServer, service)
+		})
+
+		ginkgo.It("should enforce policies to check ingress and egress policies can be controlled independently based on PodSelector [Feature:NetworkPolicy]", func() {
+			var serviceB *v1.Service
+			var podB *v1.Pod
+
+			protocolUDP := v1.ProtocolUDP
+
+			// Creating pod-b, with server container and client container,  and service-b.
+			// On receiving client request, if communication is successful, the server-b will return a file with "pod-b-server-responded"
+			podB, serviceB = createPodWithClientServerAndService(f, f.Namespace, "pod-b", []int{80}, "client-b", "pod-b-server-responded")
+			ginkgo.By("Waiting for pod-b ready", func() {
+				err := f.WaitForPodReady(podB.Name)
+				framework.ExpectNoError(err, "Error occurred while waiting for pod type: Ready.")
+			})
+
+			// Wait for pod-b with service-b to be ready
+			err := e2epod.WaitForPodRunningInNamespace(f.ClientSet, podB)
+			framework.ExpectNoError(err, "Error occurred while waiting for pod status in namespace: Running.")
+
+			defer cleanupServerPodAndService(f, podB, serviceB)
+
+			ginkgo.By("Test to check communication from pod-b to pod-a, before applying policies, which should be successful.")
+			shell := "/bin/sh"
+			opt := "-c"
+
+			// Creating command to check communication from pod-b to pod-a
+			// Before applying policy, this communication should be successful
+			targetA := fmt.Sprintf("%s.%s", service.Name, service.Namespace)
+			cmdB := fmt.Sprintf("wget -T 2 %v -O - | grep pod-a-server-responded && exit 0 || exit 1", targetA)
+
+			stdoutB, err := framework.RunKubectl(
+				"exec",
+				fmt.Sprintf("--namespace=%v", podB.Namespace),
+				podB.Name, "-c", "client-b",
+				"--",
+				shell,
+				opt,
+				cmdB)
+
+			framework.ExpectNoError(err, "Error occurred while trying to run \"kubectl exec\" in client-b.")
+			// Trimming leading and trailing spaces and newlines.
+			outputB := strings.TrimSpace(stdoutB)
+			gomega.Expect(outputB).To(gomega.Equal("pod-a-server-responded"))
```

Comment:
We can use `framework.ExpectEqual(outputB, "pod-a-server-responded")` here instead then we can remove `gomega` import.